### PR TITLE
Dotnet v6 install

### PR DESCRIPTION
**Pull request by partouf**

### CODE REVIEW

**Comment by hez2010** [comment]
The target has been bumped to v6.0.1 in https://github.com/compiler-explorer/compiler-explorer/pull/3168. Also, I added `trunk` target support in that PR.

File: bin/yaml/dotnet.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+compilers:
+  dotnet:
+    type: s3tarballs
+    dir: dotnet-{name}
+    untar_dir: artifacts
+    check_exe: bin/coreclr/Linux.x64.Checked/.dotnet/dotnet --version
+    targets:
+      - v6.0.0
```
Comment (hez2010) [inline bin/yaml/dotnet.yaml:8]:
```suggestion
      - v6.0.1
```

**Comment by hez2010** [comment]
Or maybe we can just target v6.0.0 and merge the PR first, then update it to v6.0.1 and provide trunk version in another PR later?

**Comment by partouf** [comment]
No we can do 6.0.1, I just haven't had a lot of time yet to test everything. I can get to it next week

**Comment by mattgodbolt** [comment]
Trying to move this along: is there something I can pick up on?

**Comment by partouf** [comment]
> Trying to move this along: is there something I can pick up on?

Todo:
* [x] Start a build for v6.0.1 (https://github.com/compiler-explorer/infra/actions/runs/1744697754)
* [ ] Then this needs to be merged and installed
* [ ] Then the CE PR for this needs to be merged tested on staging (though I've tested before and don't expect anything broken)

**Pull request merged**